MapboxMobileEvents/Categories/NSString+MMEVersions.m

```diff
@@ -46,6 +46,11 @@ - (NSString *)mme_semverBuildStringComponent;
 
 @implementation NSString (MMEVersions)
 
+- (NSString *)mme_stringByRemovingNonUserAgentTokenCharacters {
+return [[self componentsSeparatedByCharactersInSet:[NSCharacterSet characterSetWithCharactersInString:@" (),/:;<=>?@[]{}\"\\"]]
```

Comment:
Non-Blocker/Question:
Is there any negative side effects in removing these values? Would it cause any problems when we went to look up a given value?

If there isn't. **Ignore**. 

But if there is:
Rather than remove them, what if we encoded them like a url string? This way, we could still get any characters by reverse mapping it?

Where a space " " was mapped to a url encoded " %20"

Reply:
There *are* encoding rules for the UA string if you look at the HTTP RFC, they can be escaped with `\` to retain them, but parsing that is problematic so it's easier for us to just compress out any invalid tokens that we've given

Reply:
Makes sense. Thanks for the clarification